UPGRADE_PLAN.md

```
@@ -0,0 +1,406 @@
+
+**Location:** `src/parser/mod.rs:312-321`
+
+P2TR constructors are converted to dummy comparisons and lose semantic meaning.
+
+---
+
+## Priority 6: Code Quality
+
+### 6.1 Compiler Warnings (4)
+
+```
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)